File: Apuntes/Python/Servicios/Servidor.py
"""
TODO: Creacion de un servidor para un servicio
"""


#!/usr/bin/env python
import rospy
from std_srvs.srv import Empty, EmptyResponse #Importa el tipo de mensaje que necesitamos 

def mi_callback(request):
    """
    Funcion que sera ejecutada cuando se llame al servicio mi_servicio
    """
    print("Peticion recibida")
    return (EmptyResponse) #Retorno un mensaje vacio 

rospy.init_node('servidor') #Inicializa el nodo
mi_servicio = rospy.Service("mi_servicio", Empty, mi_callback) #Crea el servicio. Argumentos: Nombre del servicio, Tipo de mensaje, Funcion que se ejecuta cuando se recibe una peticion

rospy.spin() #Mantiene el nodo activo

File: Apuntes/C++/Servicios/Servidor.cpp
#include <ros/ros.h>
#include <Servicio/Prueba.h>
#include <iostream>
using namespace std;

bool callback (Servicio::PruebaRequest &req, Servicio::PruebaResponse &res){
    /*Funcion que se ejecuta cuando llega una peticion al servidor*/

    //Ejemplo
    ROS_INFO("Recibido: %s", req.mensaje.c_str());

    //Enviamos la respuesta
    res.valores_respuesta = 0

    return true;
}

int main(int argc, char ** argv){
    //Creacion del nodo principal
    ros::init_node("Servidor", argc, argv);
    ros::NodeHandle nh;

    //Creacion del servidor de servicios
    ros::ServiceServer server = nh.advertiseService<Servicio::PruebaRequest , Servicio:PruebaResponse >("Servidor", callback);

    //Creamos varios hilos de escucha
    ros::MultiThreadedSpinner spinner(4);
    spinner.spin();
}